src/main/java/model/TagList.java

```diff
@@ -0,0 +1,55 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class TagList {
+    private List<Tag> tags;
+
+    // Initialize with no cards
+    public TagList() {
+        this.tags = new ArrayList<>();
+    }
+
+    // Initialize from existing container of cards
+    public TagList(List<Tag> tags) {
+        this.tags = tags;
+    }
+
+    public List<Tag> getTags() {
+        return this.tags;
+    }
+
+    public void addTag(Tag tag) {
+        this.tags.add(tag);
+    }
+
+    public Tag findTag(String tagName) {
```

Comment:
Even if that is the case we still have to call `Optional<Tag>.isPresent()` to check if has an element? @ltzehan

Reply:
Yep, or restructuring our if-else into an `Option::ifPresentOrElse`. IMO this makes it clearer that the return value can be `null` because it's not obvious from the current method signature that it may not find a `Tag`. 

Especially if we call this from elsewhere, we might forget to check for nullity, so `Optional` will also help us recognize that

Reply:
Aight I'll open an issue for this